### PR TITLE
Add glossary to linting

### PR DESCRIPTION
**Pull request by ErikSchierboom**
CC @ee7

### CODE REVIEW

File: building/configlet/lint.md [hunk 1]
```diff
@@ -292,3 +292,7 @@ The `config.json` file should have the following checks:
 - The `"contributors"` values must not have duplicates
 - The `"contributors"` values are treated case-insensitively
 - Users can only be listed in either the `"authors"` or `"contributors"` array (no overlap)
+
+## Glossary
+
+- _kebab_case_: strings which characters are all in the range [a-z09-]
```
Comment (iHiD) [inline building/configlet/lint.md]:
```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9-]`
```

**Comment by ee7** [comment]
>  _kebab_case_: strings containing only characters in the range `[a-z0-9-]`

This permits strings like `-integers` and `numeric--types`.

Should the definition be a bit stricter? Maybe something like this?
`^[a-z0-9]+(-[a-z0-9]+)*$`

Note that the below doesn't match `05ab1e`, which should be a valid slug.
`^([a-z][a-z0-9]*)(-[a-z0-9]+)*$`

File: building/configlet/lint.md [hunk 2]
```diff
@@ -292,3 +292,7 @@ The `config.json` file should have the following checks:
 - The `"contributors"` values must not have duplicates
 - The `"contributors"` values are treated case-insensitively
 - Users can only be listed in either the `"authors"` or `"contributors"` array (no overlap)
+
+## Glossary
+
+- _kebab_case_: strings containing only characters in the range `[a-z0-9-]`
```
Comment (ErikSchierboom) [inline building/configlet/lint.md]:
```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9-]`; the first character must be in the range `[a-z0-9]` (no dash)

Comment (ErikSchierboom) [reply building/configlet/lint.md]: @ee7 This clear?

Comment (ee7) [reply building/configlet/lint.md]: Depends on whether you want `hello------------erik` and `hello-` to be valid slugs :)

Comment (iHiD) [reply building/configlet/lint.md]: I think we should go with the regexp`^([a-z0-9][a-z0-9]*)(-[a-z0-9]+)*$`

Comment (ErikSchierboom) [reply building/configlet/lint.md]:
```suggestion
- _kebab_case_: strings starting and ending with a character in the range `[a-z0-9]`, with zero or more characters in between in the range `[a-z0-9-]`. In regular expression format: `^[a-z0-9]+(-[a-z0-9]+)*$`
```

Comment (ErikSchierboom) [reply building/configlet/lint.md]: @iHiD @ee7 Something like this?

Comment (iHiD) [reply building/configlet/lint.md]:
```suggestion
- _kebab_case_: strings containing only characters in the range `[a-z0-9]` optionally separated by dashes (e.g. "two-fer"). In regular expression format: `^[a-z0-9]+(-[a-z0-9]+)*$`
```

**Pull request merged**